test/necb/unit_tests/tests/test_necb_skylights.rb

```diff
@@ -0,0 +1,162 @@
+            err_msg = "BTAP/OSut: Negative nomianl gross roof area (#{cas})?"
+            assert(st.osut[:gra0] > 0, err_msg)
+            err_msg = "BTAP/OSut: Negative effective gross roof area (#{cas})?"
+            assert(st.osut[:graX] > 0, err_msg)
+            err_msg = "BTAP/OSut: Missing log status (#{cas})?"
+            assert(st.osut.key?(:status), err_msg)
+            err_msg = "BTAP/OSut: Log status (#{cas})?"
+            assert(st.osut[:status].respond_to?(:to_i), err_msg)
+            err_msg = "BTAP/OSut: Missing OSut logs (#{cas})?"
+            assert(st.osut.key?(:logs), err_msg)
+            err_msg = "BTAP/OSut: OSut logs (#{cas})?"
+            assert(st.osut[:logs].is_a?(Array), err_msg)
+
+            # Tally skylight areas. Compare with GRAs.
+            skm2 = model.getSubSurfaces.sum(&:grossArea)
+            assert(skm2 > 0, "BTAP/OSut: Negative skylight area (#{cas})?")
+            gra0 = st.osut[:gra0] # gross roof area (GRA) in m2, as per SDK
+            graX = st.osut[:graX] # GRA minus overhang areas (see SmallOffice)
+
+            # The "SmallOffice" has an unconditioned (unoccupied) attic with
+            # roof overhangs. The overhanged sections of attic roof surfaces
+            # should be excluded from the calculated gross roof area, as per
+            # ASHRAE 90.1 definitions (NECB definitions are more vague). See:
+            #
+            #   github.com/rd2/osut/blob/
+            #   117c7dceb59fd8aab771da8ba672c14c97d23bd0/
+            #   lib/osut/utils.rb#L6304
+            #
+            # Relying on the total area of attic roof surfaces (for SRR%)
+            # exagerrates required skylight area targets, often by 10% to 15%.
+            # Such targets are harder to meet when dealing with skylight wells.
+            # This also leads to unfair assessments of NECB rulesets. This
+            # issue only applies for attics - not plenums.
+            #
+            # Since neither OpenStudio nor OpenStudio-Standards consider this
+            # as an issue (i.e. ASHRAE 90.1 doesn't require NECB fixed SRRs),
+            # BTAP would be perpetually 'swimming against the tide' if forcing
+            # the use of revised GRA calculations. An easier alternative is
+            # to lower proportionately the requested SRR%, while logging an
+            # informative warning to the user, e.g.:
+            #
+            #   nominal NECB2011 SRR% =  5.0%
+            #       ratio graX / gra0 = 90.0%
+            # effective NECB2011 SRR% =  4.5%
+            #
+            # EnergyPlus, OpenStudio & OpenStudio-Standards would report here a
+            # SRR% of 4.5%. The effective 'ratio' would vary based on geometry,
+            # e.g. larger building footprint, wider overhangs.
+            ratio = gra0.round > graX.round ? skm2 / graX : skm2 / gra0
+            assert(ratio.round(2) == srr, "BTAP/OSut: Incorrect SRR (#{cas})?")
```

Comment:
Although the requested SRR-based `target` area is reduced when dealing with (unconditioned) attic roof overhangs (e.g. 4.5% instead of the requested 5%), it's always possible to validate whether a requested SRR of 5% is met by dividing skylight area by the **effective** _gross roof area_.